packages/venia-ui/lib/RootComponents/Search/search.js

```diff
@@ -1,119 +1,9 @@
 
 const Search = props => {
-    const { history, location } = props;
-    const classes = mergeClasses(defaultClasses, props.classes);
-
-    // retrieve app state and action creators
-    const [appState, appApi] = useAppContext();
-    const { searchOpen } = appState;
-    const { executeSearch, toggleDrawer, toggleSearch } = appApi;
-
-    const openDrawer = useCallback(() => {
-        toggleDrawer('filter');
-    }, [toggleDrawer]);
-
-    // get the URL query parameters.
-    const urlQueryValue = getQueryParameterValue({
-        location,
-        queryParameter: 'query'
-    });
-    const categoryId = getQueryParameterValue({
-        location,
-        queryParameter: 'category'
-    });
-
-    // derive initial state from query params
-    // never re-run this effect, even if deps change
-    /* eslint-disable react-hooks/exhaustive-deps */
-    useEffect(() => {
-        // ensure search is open to begin with
-        if (toggleSearch && !searchOpen && urlQueryValue) {
-            toggleSearch();
-        }
-    }, []);
-    /* eslint-enable react-hooks/exhaustive-deps */
-
-    const apolloQueryVariable = categoryId
-        ? { inputText: urlQueryValue, categoryId }
-        : { inputText: urlQueryValue };
-
-    const { loading, error, data } = useQuery(PRODUCT_SEARCH, {
-        variables: apolloQueryVariable
-    });
-
-    if (loading) return fullPageLoadingIndicator;
-    if (error) {
-        return (
-            <div className={classes.noResult}>
-                No results found. The search term may be missing or invalid.
-            </div>
-        );
-    }
-
-    const { products } = data;
-    const { filters, total_count, items } = products;
-
-    if (items.length === 0) {
-        return <div className={classes.noResult}>No results found!</div>;
-    }
-
-    const maybeCategoryFilters = categoryId ? (
-        <CategoryFilters
-            categoryId={categoryId}
-            executeSearch={executeSearch}
-            history={history}
-            location={location}
-        />
-    ) : null;
-
-    const maybeFilterButtons = filters ? (
-        <div className={classes.headerButtons}>
-            <button onClick={openDrawer} className={classes.filterButton}>
-                Filter
-            </button>
-        </div>
-    ) : null;
-
-    const maybeFilterModal = filters ? <FilterModal filters={filters} /> : null;
-
-    return (
-        <article className={classes.root}>
-            <div className={classes.categoryTop}>
-                <div className={classes.totalPages}>
-                    {`${total_count} items`}
-                </div>
-                {maybeCategoryFilters}
-                {maybeFilterButtons}
-            </div>
-            {maybeFilterModal}
-            <section className={classes.gallery}>
-                <Gallery items={items} />
-            </section>
-        </article>
-    );
+    return <SearchPage {...props} />;
```

Comment:
Why did we need to make this a thin wrapper around the `SearchPage` component?

Reply:
> Why did we need to make this a thin wrapper around the `SearchPage` component?

I put that in the AC. Talons don't need to know about the `RootComponent` concept; ideally we'd just have all our components in `venia-ui/lib/components` and all our talons in `peregrine/lib/talons`.

This way, `Search` does the following:

- is a React component
- maps to a page type that Magento recognizes
- exists in a place the router & bundler know to look
- doesn't need a talon

And `SearchPage` does the following:

- is a React component
- renders elements
- has default styles
- has a corresponding talon with logic

Reply:
Sliding in a footnote: early on I found the naming of components, root components, and ThingPage confusing.

I tried replicating this when creating my own components/Thang, RootComponents/Thang, and components/ThangPage components — but then it looked ridiculous. I ended up reducing it all to RootComponents/Thang for each "page".

Is there an overall plan for these distinctions? I couldn't find anything around the ThingPage convention in PWA Studio docs or general React reading.

Reply:
> Is there an overall plan for these distinctions? I couldn't find anything around the ThingPage convention in PWA Studio docs or general React reading.

We're only just now getting to these root components; essentially we're starting to define that plan in this PR.

I'd like `SearchPage`, etc., to live in `components` since they're just normal components, and we can just re-export them in the `RootComponent` directory since that's how our bundler associates them with Magento routes.

You'd be hard-pressed to find similar implementations in React world since a React Router app almost always allows the client to know about the routes. The way we consult Magento for our routes and then use a bundler to map them to components from a directory is fairly unusual.

Reply:
> but then it looked ridiculous

I agree that if we have two components for one page it gets ridiculous. The `Product` and `ProductFullDetail` split is not ideal right now; logic is just arbitrarily split between the two of them, which is unfortunate.

I think with simple re-exports it won't be an issue. They're also default exports, so easily renamed on import.